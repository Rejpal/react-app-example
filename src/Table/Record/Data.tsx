import * as React from 'react'
import { RecordData } from '../../types'

interface Props {
  columns: Array<string>,
  data: RecordData,
  shouldRenderCollapser: boolean,
  onCollapserClick: () => void
}

export default function Data (props: Props) {
  const {columns, data, shouldRenderCollapser, onCollapserClick} = props

  if (!data) {
      return null;
    }
    const dataCells = columns.map((key: string) => {
      return (<td key={key}>{data[key]}</td>)
    })

    let collapser = null
    if (shouldRenderCollapser) {
      collapser = <td
        onClick={onCollapserClick}
        key="x"
        id='collapserCell'
      >x</td>
    } else {
      collapser = <td key="x" />
    }

    const result = [collapser].concat(dataCells)

    if (!result) {
      return null;
    }

    return (
      <tr>
        {result}
      </tr>
    )
}
