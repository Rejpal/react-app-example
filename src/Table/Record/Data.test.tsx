import * as React from 'react'
import Data from './Data'
import {shallow} from 'enzyme'

describe('<Data>', () => {
  const defaultProps = {
    columns: ['a', 'b', 'c'],
    data: {
      a: 'testA',
      b: 'testB',
      c: 'testC'
    },
    shouldRenderCollapser: false,
    onCollapserClick: jest.fn()
  }

  it('renders without collapser', () => {
    expect(shallow(<Data {...defaultProps} />)).toMatchSnapshot()
  })

  it('renders with collapser', () => {
    expect(shallow(<Data {...defaultProps} shouldRenderCollapser={true} />)).toMatchSnapshot()
  })

  it('propagates click on collapser', () => {
    const dummyEvent = {test: 'event'}
    const collapserClickMock = jest.fn()
    const wrapper = shallow(
      <Data
        {...defaultProps}
        shouldRenderCollapser={true}
        onCollapserClick={collapserClickMock}
      />
    )
    wrapper.find('#collapserCell').simulate('click', dummyEvent)
    expect(collapserClickMock).toHaveBeenCalledTimes(1)
    expect(collapserClickMock).toHaveBeenCalledWith(dummyEvent)
  })
})
