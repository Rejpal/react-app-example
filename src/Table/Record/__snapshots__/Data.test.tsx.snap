// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Data> renders with collapser 1`] = `
<tr>
  <td
    id="collapserCell"
    key="x"
    onClick={[MockFunction]}
  >
    x
  </td>
  <td
    key="a"
  >
    testA
  </td>
  <td
    key="b"
  >
    testB
  </td>
  <td
    key="c"
  >
    testC
  </td>
</tr>
`;

exports[`<Data> renders without collapser 1`] = `
<tr>
  <td
    key="x"
  />
  <td
    key="a"
  >
    testA
  </td>
  <td
    key="b"
  >
    testB
  </td>
  <td
    key="c"
  >
    testC
  </td>
</tr>
`;
